### PR TITLE
Linting

### PR DESCRIPTION
**Pull request by Benjaki2**
* added `gulp lint` task
* set eslint to run on `npm test`
* added travis file

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -68,5 +75,6 @@ gulp.task('clean-browser', function(cb) {
 gulp.task('watch', function() {
   gulp.watch('src/scripts/**/**/*.js', ['browser']);
 });
+//gulp.task('lint', ['lint']);
```
Comment (ZachTRice) [inline gulpfile.js]: Any reason to have this here commented out?

File: src/scripts/components/util/wv.googleAnalytics.js [hunk 2]
```diff
@@ -41,7 +41,7 @@ export default {
       ga('send', 'pageview');
     }
   },
-	/**
+  /*
 	 * @func TrackEventGA
```
Comment (ZachTRice) [inline src/scripts/components/util/wv.googleAnalytics.js]: Comment spacing is off on lines 44-56

**Comment by localjo** [comment]
We should have an .eslintrc with eslint's recommended settings. If that won't work for some reason we can do something similar to the updated eslintrc I have in my e2e branch. I'll give more details tomorrow morning.

**Review by localjo — APPROVED** [review]
Made some changes. Using eslint recommended rules in the root, fixed linting errors, including adding PropType validation.

File: package.json [hunk 3]
```diff
@@ -41,14 +40,17 @@
     "gulp-rename": "^1.2.2",
     "gulp-sourcemaps": "^1.6.0",
     "gulp-uglify": "*",
+    "vinyl-buffer": "^1.0.0",
+    "vinyl-source-stream": "^1.1.0"
+  },
+  "dependencies": {
```
Comment (Benjaki2) [inline package.json:46]: Good call separating out the dependencies from the dev dependencies @localjo

File: src/scripts/components/rangeselection/wv.timeline.dragger.js [hunk 4]
```diff
@@ -86,3 +87,21 @@ export default class TimelineDragger extends React.Component {
 TimelineDragger.defaultProps = {
   visible: true
 };
+
+TimelineDragger.propTypes = {
+  opacity: PropTypes.number,
+  startLocation: PropTypes.number,
```
Comment (Benjaki2) [inline src/scripts/components/rangeselection/wv.timeline.dragger.js:93]: @ZachTRice @tgunnoe 
How do you all feel about using the `propTypes`?
Seems like a good way to catch bugs but is also additional work.

File: gulpfile.js [hunk 5]
```diff
@@ -51,30 +48,25 @@ gulp.task('browser', function() {
...
-    .pipe(uglify().on('error', function(e){
-            console.log(e);
-         }))
+    .pipe(uglify().on('error', function(e){ console.log(e); }))
     .pipe(gulp.dest('browser'));
 });
 
 gulp.task('clean-browser', function(cb) {
   del(['./browser/*'], cb);
 });
```
Comment (Benjaki2) [inline gulpfile.js:72]: Nice cleanup of the gulpfile 👍

**Review by Benjaki2 — COMMENTED** [review]
The updates look good to me @localjo but I requested feedback from  @tgunnoe & @ZachTRice of the propTypes addition.

File: src/scripts/components/rangeselection/wv.timeline.dragger.js (hunk 4 above)
Comment (ZachTRice) [reply src/scripts/components/rangeselection/wv.timeline.dragger.js:93]: @Benjaki2 This sounds like a **really** good thing to add. Makes our props more explicit, easier to read and I also like the idea of the `isRequired` property.

Comment (ZachTRice) [reply src/scripts/components/rangeselection/wv.timeline.dragger.js:93]: And it's not much additional work. We should be defining types in our comment blocks anyway.

Comment (Benjaki2) [reply src/scripts/components/rangeselection/wv.timeline.dragger.js:93]: Cool 👌 , PR is good to go IMO.

**Comment by localjo** [comment]
Tagged v1.4.9 here so I could publish it to npm and deploy to SIT for review.

**Review by ZachTRice — APPROVED** [review]

**Pull request merged**